### PR TITLE
Enhan jump to function

### PR DESCRIPTION
**Pull request by Akarshit**
Fix for #1222

### CODE REVIEW

**Comment by mrerrormessage** [comment]
Thanks for the PR! I had one comment, but other than that, this looks terrific! Looking forward to having functions built into jump-to-declaration.

File: netlogo-gui/src/main/ide/JumpToDeclaration.scala [hunk 1]
```diff
@@ -58,15 +58,22 @@ object JumpToDeclaration {
...
+    * @param tokens list of all the tokens in the editor
+    * @return tuple of globals, owns, functions. Each value is a map
+    *         from token text to token.
+    */
+  def parseOuterScope(tokens: Seq[Token]): (Map[String, Token], Map[String, Token], Map[String, Token]) = {
     val globals = Map[String, Token]()
     val owns = Map[String, Token]()
+    val functions = Map[String, Token]()
     var i = 0
     while(i < tokens.length) {
```
Comment (mrerrormessage) [inline netlogo-gui/src/main/ide/JumpToDeclaration.scala]: This is much better. Thanks for looking at it!

**Comment by mrerrormessage** [comment]
I'd like this to be squashed and/or rebased down to a single commit, then this can be merged. Thanks again for the PR, we're very excited to have this change!

**Comment by Akarshit** [comment]
@mrerrormessage There is something wrong while building the submodule. How do I fix that in this PR?

**Comment by mrerrormessage** [comment]
It appears you have this tied to a revision of the `r` submodule which no longer exists. Hexy points to `a74e519`, so this branch should also point there.

Quick reference on updating submodule commit. Starting from NetLogo root directory:
```bash
> cd extensions/r
> git fetch
> git co a74e519
> cd ../../
> git add extensions/r
> git commit --amend # this attaches it to the prior commit, like we want for this PR. In other situations you can just use git commit normally
```

**Comment by mrerrormessage** [comment]
Looks great. Thanks!

**Pull request merged**